src/net.cpp

```diff
@@ -1034,6 +1058,12 @@ bool CConnman::AttemptToEvictConnection()
 
     if (vEvictionCandidates.empty()) return false;
 
+    // If any remaining peers are preferred for eviction consider only them.
```

Comment:
It's not entirely clear to me why we might allow preferred eviction peers to be protected under one of the above criteria -- can you elaborate on that design choice in this comment?

Reply:
Seems like fine reasoning, maybe just leave a comment to that effect so that someone could refine this later without wondering why it’s done the current way?